projects/observability/src/shared/components/bar-gauge/bar-gauge.component.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  QueryList,
+  ViewChild,
+  ViewChildren
+} from '@angular/core';
+import { DomElementMeasurerService, getPercentage, TypedSimpleChanges } from '@hypertrace/common';
+
+@Component({
+  selector: 'ht-bar-gauge',
+  styleUrls: ['./bar-gauge.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="bar-gauge">
+      <div *ngIf="this.title" class="title">{{ this.title | htDisplayTitle }}</div>
+      <div class="count">
+        {{ this.totalValue | number }} / {{ this.maxValue | number }}
+        <span class="units" *ngIf="this.units">{{ this.units }}</span>
+      </div>
+      <div class="bar">
+        <div #maxValueBar class="max-value-bar" [ngClass]="{ 'over-max-value': this.overMaxValue }">
+          <div class="segment-bars">
+            <div
+              #segmentBars
+              *ngFor="let segment of this.barSegments"
+              class="segment-bar"
+              [ngClass]="{ 'hide-divider': this.nearMaxValue }"
+              [style.background]="segment.color"
+              [style.width.%]="segment.percentage"
+            >
+              <div class="divider"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `
+})
+export class BarGaugeComponent implements OnChanges {
+  @ViewChild('maxValueBar', { read: ElementRef })
+  public maxValueBar!: ElementRef;
+
+  @ViewChildren('segmentBars', { read: ElementRef })
+  public segmentBars!: QueryList<ElementRef>;
+
+  @Input()
+  public title?: string;
+
+  @Input()
+  public units?: string;
+
+  @Input()
+  public maxValue?: number;
+
+  @Input()
+  public segments?: Segment[] = [];
+
+  public barSegments: BarSegment[] = [];
+  public totalValue: number = 0;
+  public overMaxValue: boolean = false;
+  public nearMaxValue: boolean = false;
+
+  public constructor(private readonly domElementMeasurerService: DomElementMeasurerService) {}
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.segments || changes.maxValue) {
+      this.totalValue = this.calcTotalValueFromSegments(this.segments ?? []);
+      this.barSegments = this.toBarSegments(this.segments ?? [], this.maxValue ?? 0);
+
+      this.overMaxValue = this.maxValue !== undefined && this.totalValue > this.maxValue;
+
+      this.checkNearMaxValue();
+    }
+  }
+
+  private checkNearMaxValue(): void {
```

Comment:
Does this need to re-run if dom layout changes? We could use the layout change directive for it.